src/main/java/com/atlassian/db/replica/spi/ReplicaConnectionProvider.java

```diff
@@ -0,0 +1,12 @@
+package com.atlassian.db.replica.spi;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+
+@FunctionalInterface
+public interface ReplicaConnectionProvider {
+    /**
+     * @return a connection to a replica database
+     */
+    Connection getReplicaConnection() throws SQLException;
```

Comment:
Why this is separated from ReplicaConnectionPerUrlProvider?

Reply:
I think it's a bad idea to merge it. Most of the use cases (all until now) didn't require providing a way to connect with a replica via URL.

Reply:
We also require ConnectionProvider to be implemented to use DualConnection, so the change would:
- increase the entry threshold
- break the API (I guess we are not able to provide a default implementation that would work for all the implementations in a compatible way).